lib/Sema/TypeCheckPattern.cpp

```diff
@@ -1402,8 +1402,22 @@ bool TypeChecker::coercePatternToType(Pattern *&P, DeclContext *dc, Type type,
             goto recur;
           }
 
-          diagnose(EEP->getLoc(), diag::enum_element_pattern_member_not_found,
-                   EEP->getName().str(), type);
+          auto diag = diagnose(EEP->getLoc(),
+                               diag::enum_element_pattern_member_not_found,
+                               EEP->getName().str(), type);
+
+          // If we have an optional type let's try to see if the case
+          // exists in its base type, if so we can suggest a fix-it for that.
+          if (auto *OT = dyn_cast<OptionalType>(type.getPointer())) {
```

Comment:
`OptionalType` is sugar itself, it's going to assert if I try to do `getAs` to get it.